backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/UserAuthorizationVerifier.java

```diff
@@ -36,6 +39,17 @@ public boolean userHasSiteAdminRole() {
 
     public boolean userHasPermission(UserPermission permission) {
         Optional<OrganizationRoles> orgRoles = _orgService.getCurrentOrganizationRoles();
-        return orgRoles.isPresent() && orgRoles.get().getGrantedPermissions().contains(permission);
+        // more troubleshooting help here.
+        // Note: if your not reaching this code, then grep for 'AbstractAccessDecisionManager.accessDenied' in
+        // spring library AffirmativeBased.java and set a breakpoint there.
+        if (orgRoles.isEmpty()) {
+            LOG.warn("Permission request for {} failed. No roles for org defined.", permission);
```

Comment:
This should be literally impossible. Was it happening somewhere?

Reply:
I set a breakpoint here and I think some tests triggered it. 

My biggest point of confusion is that I have to remove bob@sample.com from the `site-admin-emails` for Unit Tests to work locally, but bob@sample.com needs to be a site-admin for some local dev work (importing patients for example).

It would be really nice to have a multiple "accounts" for local dev and some way to switch between them to interactively test the different permissions.

Reply:
Ah, I see. That's ... possibly related to a discussion @jeremyzitomer-usds and I were having about keeping the dev and test configuration distinct, though I don't think it's specifically the same issue. 

That crosstalk between dev and test is explicitly bad and something I was trying to eliminate, though, so let's get together sometime soon and work out which things are failing and how to fix it.

Reply:
I think this needs a synchronous discussion :-\